sjsonnet/src-jvm/sjsonnet/SjsonnetMain.scala

```diff
@@ -159,41 +174,39 @@ object SjsonnetMain {
         }
       case (None, true) =>
         // YAML stream
+
         interp.interpret(os.read(path), OsPath(path)).flatMap {
           case arr: ujson.Arr =>
-            val renderedFiles: Seq[Either[String, String]] =
-              arr.value.toSeq.map{ v =>
-                Right(ujson.transform(v, new Renderer(indent = config.indent)).toString)
-              }
-            renderedFiles.collect{case Left(err) => err} match{
-              case Nil =>
-                val docs = renderedFiles.collect{case Right(yaml) => yaml}
-                val stream = if (docs.isEmpty) "" else docs.mkString("---\n", "\n---\n", "\n...")
-                Right[String, String](stream)
-              case errs =>
-                Left[String, String]("rendering errors:\n" + errs.mkString("\n"))
+            val stream = arr.value.toSeq match{
+              case Nil => ""
+              case Seq(single) =>
+                val renderer = rendererForConfig(new StringWriter(), config)
+                val suffix = if (isScalar(single)) "\n..." else ""
+                single.transform(renderer).toString + suffix
+              case multiple =>
+                multiple.zipWithIndex
+                  .map{
+                    case (v, i) =>
+                      val renderer = rendererForConfig(new StringWriter(), config)
+                      val rendered = v.transform(renderer).toString
+                      if (isScalar(v)) "--- " + rendered
+                      else if (i != 0) "---\n" + rendered
+                      else rendered
+                  }
+                  .mkString("\n")
             }
 
-          case _ =>
-            Left("error: stream mode: top-level object should be an array " +
-              "whose elements hold the JSON for each document in the stream.")
-        }
-      case _ =>
-        def materialize(wr: Writer) = interp.interpret0(
-          os.read(path),
-          OsPath(path),
-          new Renderer(wr, indent = config.indent)
-        )
-        config.outputFile match{
-          case None =>
-            materialize(new StringWriter).map(_.toString)
-          case Some(f) =>
-            val filePath = os.FilePath(f) match{
-              case _: os.Path => os.Path(f).relativeTo(os.pwd)
-              case _ => os.RelPath(f)
+            config.outputFile match{
```

Comment:
I didn't convert this code path to streaming output generation because it didn't come up in the profiles. We can do that in a follow up, since it's kind of orthogonal to the whole direct-YAML-generation thing

Reply:
Good idea, I'll do that